app/src/main/java/io/left/meshim/activities/ChatActivity.java

```diff
@@ -77,6 +104,47 @@ protected void onCreate(Bundle savedInstanceState) {
 
     }
 
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        switch (requestCode) {
+            case Constant.REQUEST_CODE_PICK_FILE:
+                if (resultCode == RESULT_OK) {
+                    ArrayList<NormalFile> list = data.getParcelableArrayListExtra(Constant.RESULT_PICK_FILE);
+                    // always get the first file
+                    NormalFile file = list.get(0);
+                    //getting the file name and extension of the file
+                    fileName = file.getName()+"."+getFileExtension(file.getPath());
+                    filePath = file.getPath();
+                }
+                break;
+            case Constant.REQUEST_CODE_PICK_IMAGE:
+                if (resultCode == RESULT_OK) {
+                    ArrayList<ImageFile> list = data.getParcelableArrayListExtra(Constant.RESULT_PICK_IMAGE);
+                    ImageFile imageFile= list.get(0);
```

Comment:
Are these three lines not basically the same for each case? Cane you use a generic `File` class and then in each class just set the value of the file or the Constant and re-use the code?